legend-depot-artifacts-refresh/src/main/java/org/finos/legend/depot/store/artifacts/services/ArtifactsRefreshServiceImpl.java

```diff
@@ -323,17 +323,7 @@ private MetadataEventResponse refreshVersionForProject(ProjectData project, Stri
                 if (!response.hasErrors())
                 {
                     Optional<ProjectData> latestProjectData = projects.find(project.getGroupId(), project.getArtifactId());
-                    latestProjectData.ifPresent(p ->
-                    {
-                        if (!versionId.equals(MASTER_SNAPSHOT))
-                        {
-                            p.addVersion(versionId);
-                        }
-                        p.addProperties(refreshProjectProperties(p, versionId));
-                        p.getDependencies(versionId).forEach(p::removeDependency);
-                        p.addDependencies(newDependencies);
-                        projects.createOrUpdate(p);
-                    });
+                    refreshFunctionalDependenciesOfProjectData(versionId, newDependencies, latestProjectData);
```

Comment:
this methid shold be called refresh project data... there is no dependencies here . we are refreshing the data for the project